app/middleware/simplified-account/simplified-account-strategy.middleware.js

```diff
@@ -0,0 +1,92 @@
+const { keys } = require('@govuk-pay/pay-js-commons').logging
+const { addField } = require('../../services/clients/base/request-context')
+const { getSwitchingCredentialIfExists } = require('../../utils/credentials')
+const { SERVICE_EXTERNAL_ID, ACCOUNT_TYPE } = require('../../paths').keys
+const _ = require('lodash')
+const logger = require('../../utils/logger')(__filename)
+const Connector = require('../../services/clients/connector.client.js').ConnectorClient
+const connectorClient = new Connector(process.env.CONNECTOR_URL)
+
+function getService (user, serviceExternalId, gatewayAccountId) {
+  let service
+  const serviceRoles = _.get(user, 'serviceRoles', [])
+
+  if (serviceRoles.length > 0) {
+    if (serviceExternalId) {
+      service = _.get(serviceRoles.find(serviceRole => {
+        return (serviceRole.service.externalId === serviceExternalId &&
+          (!gatewayAccountId || serviceRole.service.gatewayAccountIds.includes(String(gatewayAccountId))))
+      }), 'service')
+    }
+  }
+
+  return service
+}
+
+async function getGatewayAccountByServiceIdAndAccountType (serviceExternalId, accountType) {
+  try {
+    const params = {
+      serviceId: serviceExternalId,
+      accountType
+    }
+    let account = await connectorClient.getAccountByServiceIdAndAccountType(params)
+
+    account = _.extend({}, account, {
+      supports3ds: ['worldpay', 'stripe', 'epdq', 'smartpay'].includes(account.payment_provider),
+      disableToggle3ds: account.payment_provider === 'stripe'
+    })
+
+    const switchingCredential = getSwitchingCredentialIfExists(account)
+    const isSwitchingToStripe = switchingCredential && switchingCredential.payment_provider === 'stripe'
+    if (account.payment_provider === 'stripe' || isSwitchingToStripe) {
+      const stripeAccountSetup = await connectorClient.getStripeAccountSetup(account.gateway_account_id)
+      if (stripeAccountSetup) {
+        account.connectorGatewayAccountStripeProgress = stripeAccountSetup
+      }
+    }
+
+    return account
+  } catch (err) {
+    const logContext = {
+      error: err.message,
+      error_code: err.errorCode
+    }
+
+    if (err.errorCode === 404) {
+      logger.info('Gateway account not found', logContext)
+    } else {
+      logger.error('Error retrieving gateway account', logContext)
+    }
+  }
+}
+
+module.exports = async function getSimplifiedAccount (req, res, next) {
```

Comment:
I'm not sure about the method name `getSimplifiedAccount` here. The data model we have is simply "gateway account", not "simplified gateway account". I'd rename it to just `getGatewayAccount`.

Reply:
The name is temporary and allows us to easily differentiate between the new middleware strategy and the old one. 

I agree that it doesn't match the data model but it's important to make the differences obvious whilst we're in a transitionary phase

Reply:
Yes but the file is called "simplified-account-strategy.middleware.js". But no big deal though.

Reply:
I will find a better naming convention in the next PR